internal/cli/cmd/create/role.go

```diff
@@ -154,9 +154,20 @@ func (o *createRoleOptions) run(ctx context.Context) error {
 						Namespace: o.Project,
 						Name:      o.Name,
 					},
-					Subs:   o.Subs,
-					Emails: o.Emails,
-					Groups: o.Groups,
+					Claims: []rbacapi.Claim{
+						{
+							Name:   "subs",
+							Values: o.Subs,
+						},
+						{
+							Name:   "emails",
+							Values: o.Emails,
+						},
+						{
+							Name:   "groups",
+							Values: o.Groups,
+						},
+					},
```

Comment:
We should probably add a `--claim` flag that can be used multipe times with values of the form `<claim name>=<value>`. These should be bound to a new options field `Claims []string`. You can iterate those, parsing the entries to get your `[]rbacapi.Claim`. As a convenience for subs, emails, and groups, which are certain to be the most frequently used types of claims, you can keep those flags and keep them bound to their existing options fields. After walking the `[]string` of claims, to get `[]rbacapi.Claim`, you can walk the `Subs`, `Emails`, and `Groups` fields to amend/override what's in the `[]rbacapi.Claim`.


remove the `--sub`, `--email`, and `--group` flags in favor of a `--claim` flag that can be used multipe times with a value of the form `<claim name>=<value>`. This means the `Subs`, `Emails`, and `Groups` fields on the options struct would go away and be replaced by a `Claims []string` of `<claim name>=<value>` pairs. You can iterate over that, parsing entries to arrive at a `[]rbacapi.Claim`. This way, the CLI will work fine for

Reply:
Makes sense, I've added the flag.